provers/stark/src/constraints/evaluator.rs

```diff
@@ -184,7 +185,11 @@ impl<A: AIR> ConstraintEvaluator<A> {
 
                 // Compute all the transition constraints at this point of the LDE domain.
                 let evaluations_transition =
-                    air.compute_transition_prover(&frame, &periodic_values, rap_challenges);
+                    air.compute_transition(&TransitionEvaluationContext::Prover {
+                        frame: &frame,
+                        periodic_values: &periodic_values,
+                        rap_challenges,
+                    });
```

Comment:
very small nit, but it may be good to have two methods `TransitionEvaluationContext::new_prover_context(frame, periodic_values, rap_challenges)` and `TransitionEvaluationContext::new_verifier_context(frame, periodic_values, rap_challenges)`, (or maybe something with a better name) instantiate it before calling compute transition and then passing it. It makes a little easier to reason about it while reading the code :). Same comment as before, if this adds complexity by the compiler bothering about generics then leave it as is